### PR TITLE
Delay sn3218 import to init

### PR DESCRIPTION
**Pull request by ZodiusInfuser**

### CODE REVIEW

**Comment by Gadgetoid** [comment]
Should look at migrating this over to https://github.com/pimoroni/sn3218-python

**Comment by Gadgetoid** [comment]
To elaborate upon this, I have now release sn3218 v2.0.0 which changes the API to a class.

It includes a back-compat wrapper so it doesn't break all the libraries that currently use it (and so we have time to migrate them over to the new approach) but you should probably switch to the class-based approach in Trilobot.

Release here: https://pypi.org/project/sn3218/

You can move `import sn3218` back to the top of the file and then use `self.sn3218 = sn3218.SN3218()`. Everything else should work as-is.

File: library/trilobot/__init__.py [hunk 1]
```diff
@@ -74,6 +73,9 @@ def __init__(self):
         """ Initialise trilobot's hardware functions
         """
 
+        import sn3218
```
Comment (Gadgetoid) [inline library/trilobot/__init__.py:76]: Move back to top of file.

File: library/trilobot/__init__.py [hunk 2]
```diff
@@ -74,6 +73,9 @@ def __init__(self):
...
+        self.sn3218 = sn3218
```
Comment (Gadgetoid) [inline library/trilobot/__init__.py:77]:
```python
self.sn3218 = sn3218.SN3218()
```

**Review by Gadgetoid — CHANGES_REQUESTED** [review]
Needs to depend upon `sn3218 >= 2.0.0` then these class-based tweaks will work.

**Comment by Gadgetoid** [comment]
> it doesn't break all the libraries that currently use it

plot twist: it broke Automation HAT anyway

**Comment by ZodiusInfuser** [comment]
See https://github.com/pimoroni/trilobot-python/pull/33

**Pull request closed without merging**